src/DeveloperGame.cs

```diff
@@ -99,17 +119,32 @@ public async Task<bool> CreateNote(string content)
-        /// <param name="table_privileges"> The BigDB privileges to give this connection </param>
+        /// <param name="tablePrivileges"> The BigDB privileges to give this connection </param>
+        /// <param name="sharedSecret"> If the authentication method is Basic, the sharedSecret specified here will be used. </param>
+        [Obsolete("Please use " + nameof(CreateConnectionAsync) + ".")]
+        public void CreateConnection(string connectionId, string description, AuthenticationMethod authenticationMethod, string gameDB, List<(Table table, bool can_load_by_keys, bool can_create, bool can_load_by_indexes, bool can_delete, bool creator_has_full_rights, bool can_save)> tablePrivileges, string sharedSecret = null, CancellationToken cancellationToken = default)
+            => CreateConnectionAsync(connectionId, description, authenticationMethod, gameDB, tablePrivileges, sharedSecret, cancellationToken).GetAwaiter().GetResult();
+
+        /// <summary>
+        /// Creates a connection in the 'Settings' panel.
+        /// </summary>
+        /// <param name="connectionId"> The name of the connnection. </param>
+        /// <param name="description"> The description of the connection. </param>
+        /// <param name="authenticationMethod"> The method the connection will use for authentication. </param>
+        /// <param name="gameDB"> The database the connection has access to. By default it is set to 'Default'. If set to null, 'Default' will be used. </param>
+        /// <param name="tablePrivileges"> The BigDB privileges to give this connection </param>
         /// <param name="sharedSecret"> If the authentication method is Basic, the sharedSecret specified here will be used. </param>
-        public void CreateConnection(string connectionId, string description, AuthenticationMethod authenticationMethod, string gameDB, List<(Table table, bool can_load_by_keys, bool can_create, bool can_load_by_indexes, bool can_delete, bool creator_has_full_rights, bool can_save)> table_privileges, string sharedSecret = null)
+        public async Task CreateConnectionAsync(string connectionId, string description, AuthenticationMethod authenticationMethod, string gameDB, List<(Table table, bool can_load_by_keys, bool can_create, bool can_load_by_indexes, bool can_delete, bool creator_has_full_rights, bool can_save)> tablePrivileges, string sharedSecret = null, CancellationToken cancellationToken = default)
         {
             if (string.IsNullOrEmpty(connectionId))
                 throw new ArgumentException("Unable to create connection - connectionId cannot be null or empty.");
 
             if (connectionId.Any(char.IsUpper) || connectionId.Any(char.IsDigit))
                 throw new ArgumentException("Unable to create connection - connectionId must be lowercase and only consist of alphabetical characters.");
 
-            if (this.Connections.Any(t => t.Name == connectionId))
+            // TODO: cache Connections?
```

Comment:
I thought about this a bit earlier.
I used to use a Lazy<T> to cache everything, but reflecting upon it, I decided it'd be much less hassle overall to retrieve the latest results rather than having to expose some update methods or the like.

Reply:
Perhaps it'd be better to accept a Connection instead, rather than a `connectionId` then?

Reply:
Sure, why not.